docker/start.sh

```diff
@@ -5,10 +5,38 @@ if [ -z "$REINDEX" ]; then
 	REINDEX=10
 fi
 
+if [[ "${URL_ROOT}" = *" "* ]]; then
+	date +"%F %T Deployment path contains spaces. Deploying to root..."
+	export URL_ROOT="/"
+fi
+
+# Remove leading and trailing slashes
+URL_ROOT="${URL_ROOT#/}"
+URL_ROOT="${URL_ROOT%/}"
+
+if [ "${URL_ROOT}" = "" ]; then
+	WAR_NAME="ROOT.war"
+else
+	WAR_NAME="${URL_ROOT//\//#}.war"
+fi
+
+if [ ! -f "/usr/local/tomcat/webapps/${WAR_NAME}" ]; then
+	date +"%F %T Deployment path changed. Redeploying..."
+
+	# Delete old deployment and (re)deploy
+	rm -rf /usr/local/tomcat/webapps/*
+	opengrok-deploy -c /opengrok/etc/configuration.xml \
+            /opengrok/lib/source.war "/usr/local/tomcat/webapps/${WAR_NAME}"
+
+	# Set up redirect from /source
+	mkdir "/usr/local/tomcat/webapps/source"
```

Comment:
I guess these commands will print errors when run multiple times (e.g. stop/start/stop/start)
would it make sense to make them idempotent?

Reply:
(while above was run only once when the docker container was created)

Reply:
Like this it's only running those commands if the .war file isn't located at the expected location (i.e. first start or after URL_ROOT has been changed), isn't it? Or am I missing something?

Reply:
Looks quite okay to me with these nits:
  - if you restart the container with different `URL_ROOT` value the old one will probably be still lying around (not a big deal, just a note)
  - the exit codes of the commands are not checked (might as well run the script with `set -e`)